### PR TITLE
RELEASE_NOTES for 4.2: Gtk 3.24.15 min needed

### LINKED ISSUE #13166: half/non-working UI (workaround available)
**Describe the bug/issue**

A half working UI, where parts are not shown or parts are shown when the mousepointer moves over it to disappear again after moving away.

I see this in the terminal:

`dt_gui_load_theme: error parsing combined CSS @import url('file:///home/jade/.local/dt.dvlp/share/darktable/themes/darktable-elegant-grey.css');: darktable.css:241:23'font-feature-settings' is not a valid property name`

Followed by hundreds of GTK related error messages.

**To Reproduce**
build latest development version of darktable and start it (darktable 4.3.0+3~gd1c663b66)

**Expected behavior**
Normal looking and stable UI

**Screenshots**

scsh

and this is how it should look after starting:

scok

**Which commit introduced the error**
Don't know, haven't build/used the developers version in a while. The latest stable version (4.0.1, self build from source) does not show this behaviour.

**Platform**

* OS                : Linux (kernel 5.9.15-1)
* Distro            : Debian Buster (rel. 10.13)
* Graphics driver   : nvidia (460.39)
* OpenCL installed  : Yes (nvidia-opencl-common 460.39-1)
* OpenCL activated  : Yes
* Xorg              : 7.7+19

**Build info**

* Compiler      : clang 10.0.1 *or* gcc 8. Tried both.
* Cmake         : 3.18.4
* Glib          : 2.58.3-2
* Gtk+          : 3.24.5-1
* Exiv2         : 1.0.0.9
* Sqlite        : 3.27.2
* Optimization  : -march=native -O3 -mtune=native-pipe

**Hardware**

* CPU              : AMD Ryzen 5 3600 6-Core Processor
* GPU              : GeForce RTX 2060 SUPER 7979 MiB
* Memory           : 32 Gb + 15 Gb Swap

**Additional context**

* Reproducible with other version(s)       : No, 4.0.1, self build, works.
* Reproducible with clean config directory : Yes
* Reproducible after removing xmp file     : n/a
* Reproducible with both RAWs and JPEGs    : n/a
* Reproducible after a clean start (all dirs deleted): Yes

---

**WORK-AROUND**

It turns out that this entry in **darktable.css** is the culprit:

`font-feature-settings: "tnum";`

Commenting that one out solves the problem for me.

---

There's another issue (https://github.com/darktable-org/darktable/issues/13084) that also seems to have this entry as its culprit.

Not sure what is does, but I do not see/notice any difference after commenting it out.

**PS:** Might be related to running on an old system. I'm in dire need of a new/fresh Debian install. Just wanted to put this out here so others might find it and use the work-around provided if need be.

### PR DESCRIPTION
**Pull request by Nilvus**
Closes #13166.

Thanks to @jade-nl issue, found that new font-feature-setting: "tnum"; added is only supported since march of 2020, with Gtk+3 3.24.15. I think it remains good but actually darktable specs require only 3.22. That seems to remains good except for that new line. As 3.24.15 is now quite old, I think we could keep it as it but that means that darktable will not work correctly by default on older release.

What do we do for that? For 4.2 release, I propose to update release notes as suggested by that PR. But that's probably not enough. Or we report that line and remove it by now (it's useful but we could deal without), or we propose as a minimum 3.24.15 instead of 3.22.

@dterrahe: as that line proposal is from you, your feedback is welcomed.

### CODE REVIEW

**Comment by Nilvus** [comment]
@TurboGit: as suggested by @parafin on linked issue there, remains cmake and doc (@elstoc). Sorry for the mess and 3 commits for such thing here. I do all that directly on Github and forget some things on my way... Feel free to commit all that on one commit if needed and squash them all.

**Comment by parafin** [comment]
By doc I mostly meant README.md, I’m not sure where else we mention dependencies.

**Comment by elstoc** [comment]
Obviously it's too late to change the release (though we could push a quick 4.2.1 after) and it does seem a shame to insist on a later Gtk version for a one-line change (unless it fixes a major issue). If you plan to change the release notes, it will need rewording. Perhaps something like:

> In order to support the correct display of numbers in darktable, the minimum supported Gtk version has had to be increased to 3.24.15. For people who need to build darktable with an older version, this can be supported by reverting the following change: ...

It would be good if we could be very explicit about the issue that it fixes (I'm not sure "the correct display of numbers" is enough info) and give a link to the commit that changed it (so that people can see what changed and either manually remove it or revert the commit locally).

Alternatively, if we think that is a reasonable minimum version to support, I suggest we forego the description entirely and just state the revised dependency.

File: RELEASE_NOTES.md [hunk 1]
```diff
@@ -332,10 +332,12 @@ The following is a summary of the main features added to darktable
   formats JXL, AVIF and EXR, darktable will not include the metadata
   fields unless the user selects all of the checkboxes in the export
   preference options.
+  
+- With adding of font-feature-settings: tnum; line that allow correct display of numbers, minimal Gtk release needed (that add support of that new line) is 3.24.15. If you have older one, just remove line 241 of darktable.css file on your system. See: https://github.com/darktable-org/darktable/issues/13166
 
 ## Changed Dependencies
 
-N/A
+Bump Gtk minimum release from 3.22 to 3.24.15
```
[the remark this answers is not in the document]
Comment (Nilvus) [reply RELEASE_NOTES.md:340]: I don't know for that what to change and don't have skills and correct knowledge of all other parts to be the good one to do that. I let someone else take care of that.

**Comment by dterrahe** [comment]
> Alternatively, if we think that is a reasonable minimum version to support, I suggest we forego the description entirely and just state the revised dependency.

We've regularly seen recent "maintenance" updates to the legacy gtk 3 series introducing regressions that required workarounds but we would only see/apply these if the dev runs those versions (or if we get a very clear bug report and the reporter helps with a workaround rather than just upgrading gtk themselves). So I would be surprised if there were not a slew of other bugs with older versions of gtk, that might cause subtle or not-so-subtle but not disastrous to the point of unworkable bad, and therefore unreported, behavior. Pointing out one such issue somewhat suggests that other than that older versions of gtk are "supported" when they are _not_. I'd just update the dependency...

**Comment by Nilvus** [comment]
I just changed release notes as I think it would be good to precise that here and/or for website when we post 4.2 release on wednesday, outside packages. I let you @TurboGit decide what to do with that and update my text by @elstoc proposal. If needed, close that PR and update what's possible where it's possible. It's just to help and deal with linked issue. I hope we will have a few people with so older release anyway. 

And I agree @elstoc that it's a shame to insist on that for such minor change but that minor change with bad Gtk release make a huge impact. About commit proposal to revert, not a good thing because that commit include another fix and reverting it would break another thing (minor one). Commit is that one: https://github.com/darktable-org/darktable/commit/7b3d15b32f52706fe5db2ed852ab58e55376abf1

**Comment by Nilvus** [comment]
And so, I also agree to just bump Gtk minimal dependency and remove any description about the issue.

**Comment by elstoc** [comment]
> About commit proposal to revert, not a good thing because that commit include another fix and reverting it would break another thing (minor one)

Ok fair point. In retrospect, splitting that change into two commits would have been better than trying to fix two things at once, just to make our lives easier if we _did_ have to revert something. Commits are free

**Comment by Nilvus** [comment]
> > About commit proposal to revert, not a good thing because that commit include another fix and reverting it would break another thing (minor one)
> 
> Ok fair point. In retrospect, splitting that change into two commits would have been better than trying to fix two things at once, just to make our lives easier if we _did_ have to revert something. Commits are free

You're right but for such minor CSS adds, thinking that a revert could had been needed like what we find now, was impossible at that time.

**Comment by elstoc** [comment]
> thinking that a revert could had been needed like what we find now, was impossible at that time

Completely agree. It's never possible to be certain what unintended side-effects a change might have, but small commits that do a single thing are generally just a good habit to get into, regardless of the change you're making.

**Comment by TurboGit** [comment]
I think the current proposal is the best for 4.2 release. The implication is really bad have having a note about what is wrong and how to fix that by simply removing a CSS line seems ok to me.

@Nilvus: I'll merge that now and will copy/paste the text in the release notes for GitHub release and dtorg announce.

**Comment by elstoc** [comment]
was there something wrong with the wording I suggested? The current wording is really quite hard to follow

**Comment by TurboGit** [comment]
> was there something wrong with the wording I suggested? The current wording is really quite hard to follow

Yes, I missed it :) Now changed, thanks @elstoc !
